### PR TITLE
Support other onelogin regions

### PR DESCRIPTION
**Pull request by jspc**
The tool, currently, doesn't support onelogin EU- this PR adds that support.

It does this by taking an optional onelogin region (which defaults to US, should it not exist, for backwards compatibility) and then using this to form URLs.

The new code has 100% code coverage, including happy and unhappy path cases.

The README also reflects the change to the create provider call to include a onelogin region.

### CODE REVIEW

**Comment by johananl** [comment]
Thanks for the contribution and thorough work @jspc !
(Did you forget to include the README change?)

**Comment by johananl** [comment]
This PR addresses https://github.com/allcloud-io/clisso/issues/40.

File: cmd/providers.go [hunk 1]
```diff
@@ -28,6 +29,9 @@ func init() {
 	cmdProvidersCreateOneLogin.Flags().StringVar(&subdomain, "subdomain", "", "OneLogin subdomain")
 	cmdProvidersCreateOneLogin.Flags().StringVar(&username, "username", "",
 		"Don't ask for a username and use this instead")
+	cmdProvidersCreateOneLogin.Flags().StringVar(&region, "region", "US",
+		"Region in which onelogin API lives")
```
Comment (johananl) [inline cmd/providers.go]: I would change the flag's description to: `Region in which the OneLogin API lives` (match the capitalization of the other flags plus adding "the").

Comment (jspc) [reply cmd/providers.go]: Fair point, I'll do that

File: config/config.go [hunk 2]
```diff
@@ -22,7 +23,8 @@ func GetOneLoginProvider(p string) (*OneLoginProviderConfig, error) {
 	clientSecret := viper.GetString(fmt.Sprintf("providers.%s.client-secret", p))
 	clientID := viper.GetString(fmt.Sprintf("providers.%s.client-id", p))
 	subdomain := viper.GetString(fmt.Sprintf("providers.%s.subdomain", p))
-	user := viper.GetString(fmt.Sprintf("providers.%s.username", p))
+	username := viper.GetString(fmt.Sprintf("providers.%s.username", p))
+	region := viper.GetString(fmt.Sprintf("providers.%s.region", p))
```
Comment (johananl) [inline config/config.go]: At the moment there is no verification for the `region` flag. On the other hand, we currently have hardcoded regions in the code. I would either verify that `region` is either `EU` or `US`, **OR** decide we allow any region value and simply template this value into the URL. I think the 1st option is probably safer. At the moment I'm getting a segfault when using the new flag. Checking why.

Comment (jspc) [reply config/config.go]: In https://github.com/allcloud-io/clisso/blob/6aaee3f8aa90e2e8e63016f0af4e7519a2bab9c0/onelogin/endpoints.go#L25-L41 we verify whether the region is correct when using it, and in https://github.com/allcloud-io/clisso/blob/6aaee3f8aa90e2e8e63016f0af4e7519a2bab9c0/onelogin/endpoints.go#L25-L41 we check whether it's empty.

I suspect this case is where a region isn't set- do you get a similar segfault if, say, subdomain is empty?

File: onelogin/client.go [hunk 3]
```diff
@@ -245,13 +229,9 @@ func (c *Client) VerifyFactor(token string, p *VerifyFactorParams) (*VerifyFacto
...
-	return &Client{
-		Endpoints: Endpoints{
-			GenerateSamlAssertion: GenerateSamlAssertionURL,
-			GenerateTokens:        GenerateTokensURL,
-			GetUserByEmail:        GetUserByEmailURL,
-			VerifyFactor:          VerifyFactorURL,
-		},
-	}
+func NewClient(region string) (c *Client, err error) {
+	c.Endpoints = Endpoints{Region: region}
```
Comment (johananl) [inline onelogin/client.go]: This line causes a panic because `c` is nil.
I suggest changing the line to: `c = &Client{Endpoints: Endpoints{Region: region}}`

Comment (jspc) [reply onelogin/client.go]: Ah- I see. I don't know how I didn't hit that, I think there's a missing test on `NewClient`.

File: config/config.go (hunk 2 above)
Comment (johananl) [reply config/config.go]: Right. Well, it still might be worth doing the validation as soon as possible. I indeed get the error above when configuring a wrong region. The segfault was due to a nil pointer in `NewClient`.

File: onelogin/endpoints.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+	var base string
+	switch e.Region {
+	case "US":
+		base = usBase
+
+	case "EU":
+		base = euBase
+
+	default:
+		return fmt.Errorf("Region %q is an invalid onelogin region. Valid values are EU or US", e.Region)
```
Comment (johananl) [inline onelogin/endpoints.go]: I would fix the capitalization to `OneLogin` here, too, for uniformity. I would also add a period at the end. Sorry for being petty about these :-)

Comment (jspc) [reply onelogin/endpoints.go]: Consistency is always something to strive for, I'll add this now

File: config/config.go (hunk 2 above)
Comment (jspc) [reply config/config.go]: See latest commit:

```bash
# No region set
$(onelogin-eu) ./clisso providers create onelogin foo --client-id a --client-secret a --subdomain b
Provider 'foo' saved to config file

# Incorrect region set
$(onelogin-eu) ./clisso providers create onelogin foo2 --client-id a --client-secret a --subdomain b --region NONE
Region must be one of US or EU
```

**Comment by jspc** [comment]
@johananl ♻️

File: cmd/providers.go [hunk 5]
```diff
@@ -95,12 +99,19 @@ var cmdProvidersCreateOneLogin = &cobra.Command{
 			log.Fatalf(color.RedString("Provider '%s' already exists"), name)
 		}
 
+		switch region {
+		case "US", "EU":
+		default:
+			log.Fatal(color.RedString("Region must be one of US or EU"))
```
Comment (johananl) [inline cmd/providers.go]: Thanks. I would rephrase to `Region must be one of the following: US, EU` or `Region must be either US or EU`. Sounds better English-wise... Other than that - looks good to me.

Comment (jspc) [reply cmd/providers.go]: Seems smart, see latest
